src/drivers/cmakeDriver.ts

```diff
@@ -1286,7 +1290,7 @@ export abstract class CMakeDriver implements vscode.Disposable {
             if (this.useCMakePresets) {
                 telemetryProperties = {
                     CMakeExecutableVersion: cmakeVersion ? util.versionToString(cmakeVersion) : '',
-                    CMakeGenerator: this.getGeneratorNameForTelemetry(),
+                    CMakeGenerator: (trigger === ConfigureTrigger.taskProvider) ? presetOverride?.generator || "no.generator" : this.getGeneratorNameForTelemetry(),
```

Comment:
`presetOverride?.generator` needs to be scrubbed for PII.  Use "" instead of "no.generator" otherwise.

Reply:
Oh Ok, now I got what you mean.